compiler/src/iree/compiler/GlobalOptimization/DetachElementwiseFromNamedOps.cpp

```diff
@@ -69,6 +69,19 @@ struct DetachElementwisePattern
 
     Location loc = linalgOp.getLoc();
 
+    // Check if the output tensor access is a projected permutation
+    auto outputMap = mlir::compressUnusedDims(
+        linalgOp.getMatchingIndexingMap(outputOperands.front()));
+    if (!outputMap.isProjectedPermutation()) {
+      return rewriter.notifyMatchFailure(
+          linalgOp, "Output indexing map must be a permuted projection.");
+    }
```

Comment:
Small suggestion: `compressUnusedDims` projects out unused dims so it shouldn't affect the result of `isProjectedPermutation`.


```suggestion
    if (!linalgOp.getMatchingIndexingMap(outputOperands.front())
             .isProjectedPermutation()) {
      return rewriter.notifyMatchFailure(
          linalgOp, "Output indexing map must be a permuted projection.");
    }
```